src/app/ReadHandler.cpp

```diff
@@ -184,6 +184,13 @@ CHIP_ERROR ReadHandler::OnStatusResponse(Messaging::ExchangeContext * apExchange
                 MoveToState(HandlerState::GeneratingReports);
                 mpExchangeCtx = nullptr;
             }
+
+            //
+            // Schedule execution of the ReportingEngine to drive further report generation activity if needed
+            // on this and other subscription that have already full-filled their minimum reporting hold-off requirements
+            // (i.e OnUnblockHoldReportCallback was already called before we got this status response).
```

Comment:
So I'm trying to understand the sequence of events here, from the original issue description:

1. We send the report for the first attribute.
2. While we are waiting for a status response our hold-off timer fires.
3. We call ScheduleRun() on the reporting engine, right?
4. I guess IsReportable() returns false because we are not in the GeneratingReports state but rather in the AwaitingReportResponse state?

If so, it seems like that's the part we should document.  Something like:
```suggestion
            // It's possible that the reporting engine skipped over us because we reported
            // false from IsReportable due to not being in the GeneratingReports state.  If so,
            // we may now have become reportable and should notify the reporting engine so
            // it runs and checks whether we are in fact reportable.
```
But also, why don't we do this inside `MoveToState`, when moving to the GeneratingReports state (and maybe if IsReportable now tests true, but either way)?  Then we can remove a whole bunch of ScheduleRun calls that are scattered all over this code after `MoveToState(HandlerState::GeneratingReports);` and the scheduling will be centralized in the one place where we know we might now be reportable so it's worth trying to report us.

Reply:
Yes, your description is correct. I'll update the documentation to include your snippet.

I could certainly add that to `MoveToState`, but some of the other ones are still needed (e.g `OnUnblockHoldReportCallback`).